catalog/core/catalog-core-commands/src/test/java/org/codice/ddf/commands/catalog/RemoveCommandTest.java

```diff
@@ -99,33 +149,99 @@ protected SolrCacheMBean getCacheProxy() {
      */
     @Test
     public void testNullList() throws Exception {
-        final SolrCacheMBean mbean = mock(SolrCacheMBean.class);
-        RemoveCommand removeCommand = new RemoveCommand() {
-            @Override
-            protected SolrCacheMBean getCacheProxy() {
-                return mbean;
-            }
-        };
-
         removeCommand.ids = null;
+        removeCommand.executeWithSubject();
+        assertThat(consoleOutput.getOutput(), containsString("Nothing to remove."));
+    }
 
+    @Test
+    public void testDeleteSingleMetacardFromStore() throws Exception {
+        configureMocks(1, 0);
+        setUpRemoveCommand();
+        removeCommand.ids = ImmutableSet.of(metacardList.get(0)
+                .getId());
         removeCommand.executeWithSubject();
+        verify(catalogFrameworkMock, times(1)).delete(isA(DeleteRequest.class));
+    }
 
-        assertThat(consoleOutput.getOutput(), containsString("Nothing to remove."));
+    private void configureMocks(int... totalNumberOfMetacards) throws Exception {
```

Comment:
❗️We should add one test where the filter returns no IDs (e.g., `configureMocks(0)`).
❗️We should also have a test where both, a list of IDs and a filter are provided.